plugin/builtin/attach/attach_plugin.go

```diff
@@ -74,12 +74,24 @@ func (self *AttachPlugin) GetPanelConfig() (*plugin.PanelConfig, error) {
 					if context.Task == nil {
 						return nil, nil
 					}
-					artifactEntry, err := artifact.FindOne(artifact.ByTaskId(context.Task.Id))
+					taskId := context.Task.Id
+					if context.Task.OldTaskId != "" {
+						taskId = context.Task.OldTaskId
+					}
+
+					artifactEntry, err := artifact.FindOne(artifact.ByTaskIdAndExecution(taskId, context.Task.Execution))
```

Comment:
Can you add a test for with with/without execution find logic?

Reply:
Search the diff for "$setOnInsert". The first upsert (i.e. insert) will set the execution number.

Reply:
done.